CRM/Utils/Mail.php

```diff
@@ -314,6 +314,10 @@ public static function setEmailHeaders($params): array {
     if (!empty($params['headers'])) {
       $headers = array_merge($headers, $params['headers']);
     }
+    // Send a deprecation notice if both From and from are passed through as headers
+    if (isset($params['From']) && isset($params['from'])) {
+      CRM_Core_Error::deprecatedWarning('From and from keys in params are set this seems to be an edge case please report this in gitlab with your situation that triggered this warning.');
+    }
```

Comment:
This isn't a bad idea but I'm worried if there's a situation where you're sending 10000 emails then you'll get 10000 of these. And it's ok if they're both set, just odd if they were wildly different. So maybe let's leave it out for now?

Reply:
@demeritcowboy what if we added like a static around it if the static is set then it doesn't log it a 2nd time or something? I feel like we should probably be doing something to flag this situation as it does seem odd right?

Reply:
Maybe, but for example if the author has confusion over whether they should set From or from, then I could see it being somewhat common to set both "just in case". It's not a problem if they're set to the same value.

If we are going to warn about something, the docblock for Mail::send says `from`.

Reply:
ok have removed it now, I think we do need to try and figure out the $params['From'] is being set